### ISSUE TITLE
[BUG][code-analyzer] ERR_INVALID_ARG_TYPE Error While Running Any Code Analyzer Command In GitLab CICD Pipeline

### ISSUE THREAD

**Issue by shabu74** [forcedotcom/code-analyzer#1981]
### Have you tried to resolve this issue yourself first?

- [x] I confirm I have gone through the above steps and still have an issue to report.

### Bug Description

I have installed sf code analyzer in gitlab pipeline. I have checked the plugin after installation. Ver 5.8 is there. I run sf code-analyzer help command. It lists put all commands with details. BUT, even basic versions of all three commands - config, rules and run are getting failed with same error.

TypeError: The "path" argument must be of type string. Received undefined
    Code: ERR_INVALID_ARG_TYPE

But same command works perfectly in VS Code

### Output / Logs

```shell
(node:140) Warning: Closing file descriptor 23 on garbage collection
(Use `node --trace-warnings ...` to show where the warning was created)
    TypeError: The "path" argument must be of type string. Received undefined
    Code: ERR_INVALID_ARG_TYPE
Cleaning up project directory and file based variables
00:01
ERROR: Job failed: exit code 1
```

### Steps To Reproduce

Just run:

sf code-analyzer run

### Expected Behavior

The specified files or files under specified folder/workspace should be scanned

### Operating System

Linux (GitLab)

### Salesforce CLI Version

2.118.20

### Code Analyzer Plugin (code-analyzer) Version

5.8.0

### Node Version

20.0

### Java Version

jdk 11

### Workaround

I have tried all possible combinations of all commands

### Urgency

High

**Comment by shabu74** [comment]
@namrata111f 
I have tried the option you provided. But no difference :(

    sf plugins --core
    sf code-analyzer --help
   
    #node --trace-warnings $(which sf) code-analyzer run
    node --trace-warnings $(which sf) code-analyzer run -r eslint

(I tried both). The output is same:

@oclif/plugin-autocomplete 3.2.39 (core)
@oclif/plugin-commands 4.1.38 (core)
@oclif/plugin-help 6.2.36 (core)
@oclif/plugin-not-found 3.2.73 (core)
@oclif/plugin-plugins 5.4.54 (core)
@oclif/plugin-search 1.2.36 (core)
@oclif/plugin-update 4.7.16 (core)
@oclif/plugin-version 2.2.36 (core)
@oclif/plugin-warn-if-update-available 3.1.53 (core)
@oclif/plugin-which 3.2.42 (core)
agent 1.26.2 (core)
apex 3.9.5 (core)
api 1.3.7 (core)
auth 4.1.0 (core)
code-analyzer 5.8.0
data 4.0.67 (core)
deploy-retrieve 3.24.7 (core)
info 3.4.100 (core)
limits 3.3.74 (core)
marketplace 1.3.8 (core)
org 5.9.57 (core)
packaging 2.24.5 (core)
schema 3.3.90 (core)
settings 2.4.54 (core)
sobject 1.4.81 (core)
telemetry 3.6.71 (core)
templates 56.4.0 (core)
trust 3.7.113 (core)
user 3.6.45 (core)
Uninstalled JIT Plugins:
community 3.3.48
custom-metadata 3.3.72
dev 2.5.1
devops-center 1.2.27
flow 1.0.5
signups 2.6.56
@salesforce/sfdx-plugin-lwc-test 1.2.1
Analyze your code to ensure it adheres to best practices.
USAGE
  $ sf code-analyzer COMMAND
COMMANDS
  code-analyzer config  Output the current state of configuration for Code
                        Analyzer.
  code-analyzer rules   List the rules that are available to analyze your code.
  code-analyzer run     Analyze your code with a selection of rules to ensure
                        good coding practices.
    TypeError: The "path" argument must be of type string. Received undefined
    Code: ERR_INVALID_ARG_TYPE
Cleaning up project directory and file based variables
00:01
ERROR: Job failed: exit code 1

**Comment by namrata111f** [comment]
Can you please share your code base against which you are trying to run, I can test on that once as well? We need to be able to reproduce the issue to understand it. And also just try this command once individually without other 3 to see if it is also giving the path error:
```
sf code-analyzer run -r eslint
```
